ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/LocalTxMonitor/Server.hs

```diff
@@ -31,41 +36,43 @@ localTxMonitorServer mempool =
       { recvMsgDone = do
           pure ()
       , recvMsgAcquire = do
-          s <- atomically $ (,) <$> getCapacity mempool <*> getSnapshot mempool
+          s <- atomically query
           pure $ serverStAcquiring s
       }
 
     serverStAcquiring
-      :: (MempoolCapacityBytes, MempoolSnapshot blk)
+      :: (TxMeasure blk, MempoolSnapshot blk)
       -> ServerStAcquiring (GenTxId blk) (GenTx blk) SlotNo m ()
     serverStAcquiring s@(_, snapshot) =
-      SendMsgAcquired (snapshotSlotNo snapshot) (serverStAcquired s (snapshotTxs snapshot))
+      SendMsgAcquired
+        (snapshotSlotNo snapshot)
+        (serverStAcquired s (snapshotTxs snapshot))
 
     serverStAcquired
-      :: (MempoolCapacityBytes, MempoolSnapshot blk)
-      -> [(Validated (GenTx blk), idx)]
+      :: (TxMeasure blk, MempoolSnapshot blk)
+      -> [TxTicket (TxMeasure blk) (Validated (GenTx blk))]
       -> ServerStAcquired (GenTxId blk) (GenTx blk) SlotNo m ()
     serverStAcquired s@(capacity, snapshot) txs =
       ServerStAcquired
       { recvMsgNextTx =
           case txs of
             []  ->
               pure $ SendMsgReplyNextTx Nothing (serverStAcquired s [])
-            (txForgetValidated -> h, _):q ->
+            (txTicketTx -> txForgetValidated -> h):q ->
               pure $ SendMsgReplyNextTx (Just h) (serverStAcquired s q)
       , recvMsgHasTx = \txid ->
           pure $ SendMsgReplyHasTx (snapshotHasTx snapshot txid) (serverStAcquired s txs)
       , recvMsgGetSizes = do
-          let MempoolSize{msNumTxs,msNumBytes} = snapshotMempoolSize snapshot
+          let MempoolSize{msNumTxs,msSize} = snapshotMempoolSize snapshot
           let sizes = MempoolSizeAndCapacity
-                { capacityInBytes = getMempoolCapacityBytes capacity
-                , sizeInBytes     = msNumBytes
+                { capacityInBytes = getSizeInBytes $ txMeasureBytes snapshot capacity
+                , sizeInBytes     = getSizeInBytes $ txMeasureBytes snapshot msSize
                 , numberOfTxs     = msNumTxs
                 }
```

Comment:
Shall we (create an issue to) enrich the LocalTxMonitor API here to return the full-dimensional capacity (as a new request/response pair probably)?

Reply:
Right, it wasn't clear to me.

@KtorZ does the use case for `LocalTxMonitor` care about the _other_ sizes of the mempool? EG total script ExUnits, etc --- beyond just the byte size?

Reply:
For context: This PR changes the mempool capacity to not only look at tx size, but also at the other measures, like script ExUnits and (starting with Babbage) reference scripts size. So a mempool might be almost full w.r.t. tx byte size, but almost empty w.r.t. reference scripts size. 

Concretely, adding a tx to the mempool will not block iff forging two blocks back-to-back now with the current mempool contents plus the new tx would result in an empty mempool.

Reply:
It definitely does @nfrisby, since the main use case people have for looking at the size is knowing if they can fit one more transaction. And the transaction may overflow via any of the dimension. If we could eventually expose those other dimensions, that'd be great!

Reply:
Thanks for the input! I opened https://github.com/IntersectMBO/ouroboros-consensus/issues/1178